src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs

```diff
@@ -37,13 +37,20 @@ public static XmlDocumentationProvider CreateFromBytes(byte[] xmlDocCommentBytes
             return new ContentBasedXmlDocumentationProvider(xmlDocCommentBytes);
         }
 
+        private static XmlDocumentationProvider DefaultXmlDocumentationProvider { get; } = new NullXmlDocumentationProvider();
+
         /// <summary>
         /// Creates an <see cref="XmlDocumentationProvider"/> from an XML documentation file.
         /// </summary>
         /// <param name="xmlDocCommentFilePath">The path to the XML file.</param>
         /// <returns>An <see cref="XmlDocumentationProvider"/>.</returns>
         public static XmlDocumentationProvider CreateFromFile(string xmlDocCommentFilePath)
         {
+            if (!File.Exists(xmlDocCommentFilePath))
```

Comment:
Here's the caller: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Host/Documentation/DocumentationProviderServiceFactory.cs,34

How would you suggest we change this code?

Reply:
Dunno, now that you're asking me to look further I only see more problems. :-)

1. If the file exists, that's called, and then deleted, will we still crash once we actually try to read the file? The read is not happening "near" in time to the File.Exists, but a fair bit later. Not that a File.Exists may not be appropriate to avoid a first-chance exception, but this isn't adding the resiliency this change might need.
2. This seems to be a cache with no expiration policy. I was hoping this is only used for people who are doing their own workspaces, but it looks like people with their own workspaces would also get this inside Visual Studio, which is Most Bad™ since it should be forwarding to the VS member index service. The odd bit is we do have the code that does it, but it doesn't seem to implement itself as a workspace service...

Reply:
Right, see my comment here: https://github.com/dotnet/roslyn/pull/32005#issuecomment-449618678 regarding the memory leak. You may consider opening a new issue to fix that and the other concern that you have with file existence.

I did verify that the fix above is a significant improvement for us (and we don't need 100%, it's enough that it avoids the exception in 99.5% of cases). So yes, we can make this even better but that's no longer blocking us ;)

Reply:
Well, I guess the main thing I'm worried about is that this isn't being replaced by the VS implementation, so if we're using this in VS for Windows, we have even bigger issues.